### DISCUSSION TITLE
How to deploy to ACA application using Dapr

### DISCUSSION THREAD

**Discussion by laurentkempe** [microsoft/aspire#852] [Q&A]
I updated the template to use Dapr and it work perfectly locally. Trying to update my application deployed to Azure Container Apps I get an error while running `azd deploy`

> ERROR: failed deploying service 'apiservice': failing invoking action 'deploy', generating container app manifest: unsupported resource type: executable.v0

Is there something I can do? Is this scenario already supported?

I tried to update to Azd 1.5.0 and now I get this

```
 ❯  azd deploy

Deploying services (azd deploy)

  |    ===| Deploying service apiservice (Pushing container image)panic: send on closed channel

goroutine 78 [running]:
github.com/azure/azure-dev/cli/azd/pkg/async.(*TaskContextWithProgress[...]).SetProgress(...)
        github.com/azure/azure-dev/cli/azd/pkg/async/task_context.go:67
github.com/azure/azure-dev/cli/azd/pkg/project.syncProgress[...](0xc000138330, 0xc00052c001)
        github.com/azure/azure-dev/cli/azd/pkg/project/service_manager.go:656 +0xbd
created by github.com/azure/azure-dev/cli/azd/pkg/project.runCommand[...].func1 in goroutine 91
        github.com/azure/azure-dev/cli/azd/pkg/project/service_manager.go:636 +0xb9
```

**Comment by davidfowl** [comment]
Doesn't work yet:

https://github.com/dotnet/aspire/pull/862

**Reply by rithinch** [reply]
I'm having the same issue, using Aspire 9 and Azd CLI 1.11.0 - dapr with custom components works perfectly locally. But, when deploying to azure, the container apps `dapr` section seems to be disabled. Any ideas?

**Reply by davidfowl** [reply]
In pretty sure you can deploy custom dapr components into ACA

**Comment by philliphoff** [comment]
Just a note that it also requires changes on the Azure Dev CLI side: https://github.com/Azure/azure-dev/pull/2992

Note that this is an incremental step towards supporting basic Dapr scenarios (e.g. service invocation and default state stores and pub-sub) but won't support every combination of Dapr components.  Service bindings and specific state stores will still need some manual configuration post-deployment.  (Would love input on how one imagines that would work.)

**Reply by laurentkempe** [reply]
My simple use case using Dapr state store works 👍🏼
